pkg/server/singleprocess/service_pipeline.go

```diff
@@ -390,6 +396,17 @@ func (s *Service) buildStepJobs(
 				return nil, nil, nil, err
 			}
 
+			// Add the parent step workspace ref.
+			// **Note** this only works now because we are limiting nested
+			// pipelines to 1 layer.
```

Comment:
A quick note about this - these functions are written in such a way that embedded pipelines could be infinitely nested. The 1 layer of pipeline stanza nesting is mostly an artificial limitation of our parser, but users could still use pipeline step refs to make more complicated references that go beyond a single layer. Since this function is recursive I think this approach will work regardless of the limit, so maybe the comment could be something like:

```suggestion
			// Add the parent step workspace ref and apply it to all embedded
			// pipeline job templates if step was not configured with a
			// workspace ref.
```

Reply:
Sounds reasonable, thanks